services/QuillLMS/app/lib/quill_authentication.rb

```diff
@@ -81,6 +95,14 @@ def previewing_student_dashboard?
     !session[:preview_student_id].nil?
   end
 
+  def viewing_demo_account?
+    !session[:demo_id].nil?
```

Comment:
This is super-minor, but it might be cleaner to just make this `session[:demo_id].present?`.  Easier to parse without the `!` if it works here.